clang/lib/Format/Format.cpp

```diff
@@ -3246,8 +3246,15 @@ tooling::Replacements sortCppIncludes(const FormatStyle &Style, StringRef Code,
   SmallVector<StringRef, 2> RawStringMatches;
   std::string RawStringTermination = ")\"";
 
-  for (;;) {
-    auto Pos = Code.find('\n', SearchFrom);
+  for (const auto Size = Code.size(); SearchFrom < Size;) {
```

Comment:
Why is the extra `Size` variable needed? Why can't it stay as `for (;;)` ?

Reply:
Because it's used in `SearchFrom < Size`. My previous version using `for (;;)` caused a crash in ToolingTests.